tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/DockerUtils.java

```diff
@@ -1,15 +1,43 @@
 package org.graalvm.internal.tck;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
 import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class DockerUtils {
 
-    public static Set<String> getAllowedImages() throws IOException {
-        return new HashSet<>(Files.readAllLines(Paths.get("./tests/tck-build-logic/src/main/resources/AllowedDockerImages.txt")));
+    public static Set<String> getAllowedImages() throws IOException, URISyntaxException {
+        String dockerfileDirectory = Paths.get("./tests/tck-build-logic/src/main/resources/allowed-docker-images").toString();
+        File[] dockerFiles = new File(dockerfileDirectory).listFiles();
+        if (dockerFiles == null) {
+            throw new RuntimeException("Cannot find allowed-docker-images directory content");
+        }
+
+        final String FROM = "FROM";
+        Set<String> allowedImages = new HashSet<>();
+        for (File dockerFile : dockerFiles) {
+            BufferedReader file = new BufferedReader(new FileReader(dockerFile));
+            String image = file.readLine().substring(FROM.length()).trim();
+
+            if (file.readLine() != null) {
```

Comment:
Mh, this seems a bit too strict. What if we want to allow `org.opencontainers.image.source` labels in the future? What if you just adjust the previous version to something like this:

```java
List<String> images = Files.readAllLines(dockerFile.toPath())
                    .stream()
                    .filter(line -> line.startsWith(FROM))
                    .map(line -> line.substring(FROM.length()).trim())
                    .toList();
if (images.size() != 1) {
    throw new RuntimeException("Dockerfile: " + dockerFile.getName() + " must contain only one FROM line, got '" + images.size() + "' (" + images+ "). Please read our documentation: ...";
}
```

Reply:
Sure, that is also an option